### ISSUE TITLE
Returns 404 for valid endpoints if it's not fully loaded

### ISSUE THREAD

**Issue by aramirez-es** [scm-spain/karyon-rest-router#20]
Since last update, the behavior changed a bit, as described here:

**Previous to 1.4.1**
- First run the app
```bash
$ gradlew clean runApp
. . .
2015-11-30 11:27:18 INFO  log:67 - Started SocketConnector@0.0.0.0:8077
```
- At this point if you request a valid endpoint, that **request will be blocked** until the whole server to be up. Then the response is the expected.
- Following requests will be server without blocking.

**From 1.4.1 on**
- First run the app
```bash
$ gradlew clean runApp
. . .
2015-11-30 11:27:18 INFO  log:67 - Started SocketConnector@0.0.0.0:8077
```
- At this point if you request a valid endpoint, that **request will return a 404** since the whole server is not up yet.
- Following requests will be server as expected.

**Comment by fiunchinho** [comment]
I think it shouldn't return a 404. Previous behavior was the expected.

**Comment by aramirez-es** [comment]
I agree @fiunchinho, indeed it may cause a undesired behavior in a context of: high concurrency, continuous delivery and micro caching. 404 may be cached for a while and lot of requests to get a unexpected response.

**Comment by padilo** [comment]
I agree too, It should work as previous versions.

While looking to solve this, It seems that now the package property is required, if not specified it doesn't register any endpoint.

To workaround this just set this the property 'com.scmspain.karyon.rest.property.packages' with the package to look for endpoints.

Please, change the label to bug :)

**Comment by padilo** [comment]
I think now is solved, if someone can check it or thinks is solved please close this issue
